ocaml/idl/datamodel.ml

```diff
@@ -6577,6 +6579,25 @@ let vdi_disable_cbt = call
     ~allowed_roles:_R_VM_ADMIN
     ()
 
+let vdi_hidden_cbt_setter = call
+    ~name:"hidden_cbt_setter"
+    ~in_oss_since:None
+    ~in_product_since:rel_inverness
+    ~params:[Ref _vdi, "self", "The VDI for which CBT enabled status should be set";
+             Bool, "value", "The value to set"]
+    ~errs:[
+      Api_errors.sr_operation_not_supported;
+      Api_errors.vdi_missing;
+      Api_errors.sr_not_attached;
+      Api_errors.sr_no_pbds;
+      Api_errors.operation_not_allowed;
+      Api_errors.vdi_incompatible_type;
+      Api_errors.vdi_on_boot_mode_incompatible_with_operation;
+    ]
+    ~hide_from_docs:true
+    ~allowed_roles:_R_POOL_ADMIN
```

Comment:
I'm not sure what privileges SM has, but we should set this field to exactly that.

Reply:
I will check with Chandrika

Reply:
This looks good, as it's already the highest privilege level. So if you tried it and it worked, then there is no need to change it :smile:

Reply:
Wait, I was wrong, actually there is a higher role, `_R_LOCAL_ROOT_ONLY`, let's try that and see if it still works.

Reply:
well my tests were just running the API from dom0, hopefully that is sufficient

Reply:
Yes, that's perfect, because that's where SM runs too.